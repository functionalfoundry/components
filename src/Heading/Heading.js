/* @flow */
import React from 'react'
import Theme from 'js-theme'
import {
  Fonts,
} from '@workflo/styles'
import Text from '../Text'

type PropsT = {
  size: 'Tiny' | 'Small' | 'Base' | 'Large' | 'Huge',
  testId?: string,
  theme: Object,
}

const defaultProps = {
  size: 'Base',
  theme: {},
}

const Heading = ({
  size,
  testId, // eslint-disable-line no-unused-vars
  theme,
  ...props
}: PropsT) => (
  <Text
    {...theme.heading}
    {...props}
    size={size}
    element={elementMap[size]}
  />
)

Heading.defaultProps = defaultProps

const defaultTheme = ({
  size,
}: PropsT) => ({
  heading: {
    ...Fonts[size.toLowerCase()],
  },
})

const elementMap = {
  Tiny: 'h5',
  Small: 'h4',
  Base: 'h3',
  Large: 'h2',
  Huge: 'h1',
}

export default Theme('Heading', defaultTheme)(Heading)
