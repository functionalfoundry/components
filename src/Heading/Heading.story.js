import React from 'react'
import { storiesOf } from '@kadira/storybook'
import Heading from './Heading'

storiesOf('Heading', module)
  .add('Sizes', () => (
    <div
      style={styles.container}
    >
      <Heading
        size='huge'
      >
        {'huge'}
      </Heading>
      <Heading
        size='large'
      >
        {'large'}
      </Heading>
      <Heading
        size='base'
      >
        {'base'}
      </Heading>
      <Heading
        size='small'
      >
        {'small'}
      </Heading>
      <Heading
        size='tiny'
      >
        {'tiny'}
      </Heading>
    </div>
  ))

const styles = {
  container: {
    width: '100%',
  },
}
