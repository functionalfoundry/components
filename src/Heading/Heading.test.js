import React from 'react'
import renderer from 'react-test-renderer'
import Heading from '.'

test('Renders without props', () => {
  const heading = renderer.create(<Heading size="small" />).toJSON()
  expect(heading).toMatchSnapshot()
})
