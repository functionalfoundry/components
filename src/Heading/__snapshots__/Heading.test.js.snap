exports[`test Renders without props 1`] = `
<div
  className="view_1ps7m7i heading_1v6ynb9 text_1v6ynb9"
  data-test-id={undefined}
  size="small" />
`;
